src/directConnectManager.ts

```diff
@@ -170,16 +179,16 @@ export class DirectConnectionManager {
     // combine the returned ConnectionSpec with the CustomConnectionSpec before storing
     // (spec comes first because the ConnectionSpec will try to override `id` as a string)
     const mergedSpec: CustomConnectionSpec = {
-      ...connection.spec,
-      id: spec.id,
-      formConnectionType: spec.formConnectionType,
+      ...updatedSpec,
```

Comment:
`connection` has passwords starred out, so we save what we sent to the sidecar instead